### PR TITLE
fix(avo-2742): fix duplicate collections which caused multiple stacking dropdowns

### PR DESCRIPTION
**Pull request by SHDBramVandenbussche**
[AVO-2741](https://meemoo.atlassian.net/browse/AVO-2741)
[AVO-2742](https://meemoo.atlassian.net/browse/AVO-2742)
[AVO-2743](https://meemoo.atlassian.net/browse/AVO-2743)
[AVO-2727](https://meemoo.atlassian.net/browse/AVO-2727)

[Avo2-proxy PR](https://github.com/viaacode/avo2-proxy/pull/656)
[Avo2-proxy PR](https://github.com/viaacode/avo2-proxy/pull/659)

Before:
image

After: 
image

[AVO-2742]: https://meemoo.atlassian.net/browse/AVO-2742?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AVO-2727]: https://meemoo.atlassian.net/browse/AVO-2727?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[AVO-2741]: https://meemoo.atlassian.net/browse/AVO-2741?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AVO-2743]: https://meemoo.atlassian.net/browse/AVO-2743?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by WardVerc1 — APPROVED** [review]

File: src/collection/components/modals/DeleteCollectionModal.tsx [hunk 1]
```diff
@@ -30,6 +30,12 @@ const DeleteCollectionModal: FunctionComponent<DeleteCollectionModalProps> = ({
 
 	const renderDeleteMessageParagraph = () => {
 		if (isSharedWithOthers) {
+			if (contributorCount === 1) {
+				return tHtml(
+					'Ben je zeker dat je jezelf van deze collectie wil wissen? Deze opdracht is met 1 andere persoon gedeeld. Deze verliezen dan toegang.'
```
Comment (WardVerc1) [inline src/collection/components/modals/DeleteCollectionModal.tsx:35]: Isn't the owner also a contributor? So if it is shared with 1 person, the count would be 2

**Review by WardVerc1 — APPROVED** [review]

File: src/assignment/modals/DeleteAssignmentModal.tsx [hunk 2]
```diff
@@ -30,6 +34,11 @@ const DeleteAssignmentModal: FunctionComponent<DeleteAssignmentModalProps> = ({
 
 	const renderDeleteMessageParagraph = () => {
 		if (isSharedWithOthers) {
+			if (contributorCount === 1) {
+				return tHtml(
+					'Ben je zeker dat je jezelf van deze opdracht wil wissen? Deze opdracht is met 1 andere persoon gedeeld. Deze verliezen dan toegang.'
```
[the remark this answers is not in the document]
Comment (bertyhell) [reply src/assignment/modals/DeleteAssignmentModal.tsx:39]: no, the owner is separate under the owner_profile_id column (owner relationship)
only contributors wirth type CONTRIBUTER and VIEWER are in the contributors table

File: src/collection/components/modals/DeleteCollectionModal.tsx (hunk 1 above)
Comment (bertyhell) [reply src/collection/components/modals/DeleteCollectionModal.tsx:35]: no, the owner is separate under the owner_profile_id column (owner relationship)
only contributors wirth type CONTRIBUTER and VIEWER are in the contributors table

File: src/assignment/modals/DeleteAssignmentModal.tsx [hunk 3]
```diff
@@ -20,6 +22,8 @@ const DeleteAssignmentModal: FunctionComponent<DeleteAssignmentModalProps> = ({
 	isContributor,
 	isSharedWithOthers,
 	contributorCount,
+	responses = false,
```
Comment (bertyhell) [inline src/assignment/modals/DeleteAssignmentModal.tsx:25]:
```suggestion
	hasResponses = false,
```

File: src/assignment/modals/DeleteAssignmentModal.tsx [hunk 4]
```diff
@@ -20,6 +22,8 @@ const DeleteAssignmentModal: FunctionComponent<DeleteAssignmentModalProps> = ({
...
+	buildType = false,
```
Comment (bertyhell) [inline src/assignment/modals/DeleteAssignmentModal.tsx:26]:
```suggestion
	containsBuildBlock = false,
```

File: src/assignment/modals/DeleteAssignmentModal.tsx [hunk 5]
```diff
@@ -42,6 +51,18 @@ const DeleteAssignmentModal: FunctionComponent<DeleteAssignmentModalProps> = ({
...
+			return tHtml(
+				'assignment/views/assignment-overview___deze-opdracht-bevat-mogelijk-collecties-die-eveneens-verwijderd-zullen-worden'
+			);
+		}
+
+		if (responses) {
+			return tHtml(
+				'assignment/views/assignment-overview___leerlingen-bekeken-deze-opdracht-reeds'
+			);
+		}
```
Comment (bertyhell) [inline src/assignment/modals/DeleteAssignmentModal.tsx:64]: what if the assignment is shared and has responses?
Then the user only sees the contributor warning message?

maybe you'll have to change this to:
```typescript
let contributorMessage = null;
let responsesMessages = null;

if (isSharedWithOthers) {
	if (contributorCount === 1) {
		contributorMessage = tHtml(
			'Ben je zeker dat je jezelf van deze opdracht wil wissen? Deze opdracht is met 1 andere persoon gedeeld. Deze verliezen dan toegang.'
		);
	} else {
		contributorMessage  = tHtml(
			'assignment/modals/delete-assignment-modal___ben-je-zeker-dat-je-jezelf-van-deze-opdracht-wil-wissen-deze-opdracht-is-met-count-andere-mensen-gedeeld-deze-verliezen-dan-toegang',
			{ count: contributorCount }
		);
	}
} else if (isContributor) {
	contributorMessage  = tHtml(
		'assignment/modals/delete-assignment-modal___ben-je-zeker-dat-je-jezelf-van-deze-opdracht-wil-wissen'
	);
}

if (responses) {
	if (buildType) {
		responsesMessages = tHtml(
			'assignment/views/assignment-overview___deze-opdracht-bevat-mogelijk-collecties-die-eveneens-verwijderd-zullen-worden'
		);
	} else {
		responsesMessages  = tHtml(
			'assignment/views/assignment-overview___leerlingen-bekeken-deze-opdracht-reeds'
		);
	}
}

return <>
	{contributorMessage}
	{responsesMessages}
</>
```

File: src/assignment/views/AssignmentOverview.tsx [hunk 6]
```diff
@@ -411,15 +415,18 @@ const AssignmentOverview: FunctionComponent<AssignmentOverviewProps> = ({
...
+				contributorObj?.id,
+				user.profile?.id
+			);
 		} else {
 			await deleteAssignment(markedAssignment?.id, user);
 		}
 
 		handleDeleteModalClose();
 
 		await updateAndReset();
```
Comment (bertyhell) [inline src/assignment/views/AssignmentOverview.tsx:429]: add a try catch to add a toast message if the delete fails

File: src/collection/queries/get-collections-by-owner.graphql [hunk 7]
```diff
@@ -1,5 +1,5 @@
 query getCollectionsByOwner(
```
Comment (bertyhell) [inline src/collection/queries/get-collections-by-owner.graphql:1]: also update the query name and the file name so it is more clear that the collaborator id that is passed can be an owner or a contributor
getCollectionsByOwnerOrContributor

**Review by bertyhell — CHANGES_REQUESTED** [review]

**Pull request merged**